frontend/workflows/projectSelector/src/helpers.tsx

```diff
@@ -96,7 +97,92 @@ const exclusiveProjectDependencies = (
   return { upstreams, downstreams };
 };
 
+// evaluates whether an upstream/downstream should be hidden based on the checked status of/exclusivity to a project(s)
+const hidden = (
```

Comment:
seems like it would be more efficient figure out the hidden projects (or conversely the ones that should be displayed) once for all selected projects rather than per project, lots of nested loops happening.

Reply:
thx for raising! updated to remove unnecessary extra looping